cranelift/codegen/src/isa/aarch64/mod.rs

```diff
@@ -32,10 +32,14 @@ impl AArch64Backend {
         AArch64Backend { triple, flags }
     }
 
-    fn compile_vcode(&self, func: &Function, flags: &settings::Flags) -> VCode<inst::Inst> {
-        // This performs lowering to VCode, register-allocates the code, computes
-        // block layout and finalizes branches. The result is ready for binary emission.
-        let abi = Box::new(abi::AArch64ABIBody::new(func));
+    /// This performs lowering to VCode, register-allocates the code, computes block layout and
+    /// finalizes branches. The result is ready for binary emission.
+    fn compile_vcode<'flags>(
```

Comment:
I'm not super-excited about the `'flags` lifetime carried into the `VCode`; it will propagate to anything else that contains a `VCode` which is not very ergonomic. (On preview of the rest of the patch below, it already led to a lot of boilerplate churn.) Reading the generated code, it looks like it's currently a `[u8; 7]` (so the pointer is larger than the flags array itself!). In general, with a bit per flag, I wouldn't expect it to grow beyond a few words, for a while anyway. Could we just clone it instead?

Reply:
Oh yes, it's much simpler.